src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/ContentPickerFieldSettingsDriver.cs

```diff
@@ -1,34 +1,107 @@
 using System;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
 using OrchardCore.ContentFields.Fields;
 using OrchardCore.ContentManagement.Metadata.Models;
 using OrchardCore.ContentTypes.Editors;
+using OrchardCore.DisplayManagement.ModelBinding;
 using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Mvc.ModelBinding;
 
 namespace OrchardCore.ContentFields.Settings
 {
     public class ContentPickerFieldSettingsDriver : ContentPartFieldDefinitionDisplayDriver<ContentPickerField>
     {
+        public const string ContentTypes = "ContentTypes";
+        public const string Stereotypes = "Stereotypes";
+        public const string AllTypes = "AllTypes";
+
+        private readonly IStringLocalizer S;
+
+        public ContentPickerFieldSettingsDriver(IStringLocalizer<ContentPickerFieldSettingsDriver> stringLocalizer)
+        {
+            S = stringLocalizer;
+        }
+
         public override IDisplayResult Edit(ContentPartFieldDefinition partFieldDefinition)
         {
-            return Initialize<ContentPickerFieldSettings>("ContentPickerFieldSettings_Edit", model => partFieldDefinition.PopulateSettings(model))
-                .Location("Content");
+            return Initialize<ContentPickerFieldSettingsViewModel>("ContentPickerFieldSettings_Edit", model =>
+            {
+                var settings = partFieldDefinition.GetSettings<ContentPickerFieldSettings>();
+                model.Hint = settings.Hint;
+                model.Required = settings.Required;
+                model.Multiple = settings.Multiple;
+                model.DisplayedContentTypes = settings.DisplayedContentTypes;
+                model.DisplayedStereotypes = settings.DisplayedStereotypes;
+                model.Source = GetSource(settings);
+                model.Stereotypes = String.Join(',', settings.DisplayedStereotypes ?? Array.Empty<string>());
```

Comment:
If `DisplayedContentTypes` doesn't need a backing field then `DisplayedStereotypes` should (could) not either.

Reply:
Done